doc/class_property_descriptor.md

```diff
@@ -31,8 +30,9 @@ Napi::Object Example::Init(Napi::Env env, Napi::Object exports) {
         InstanceAccessor<&Example::GetValue>("readOnlyProp")
     });
 
-    constructor = Napi::Persistent(func);
-    constructor.SuppressDestruct();
+    Napi::FunctionReference *constructor = new Napi::FunctionReference();
+    *constructor = Napi::Persistent(func);
+    env.SetInstanceData(constructor);
```

Comment:
A finalizer is required for this FunctionReference pointer right?

Reply:
Hi @legendecas ,

Good question... I was under the impression that the instance data would be `delete`d, so there would be no need for a Finalizer.
- In the case of multiple calls to `Example::Init`, the previous instance data would be overwritten via most recent call to `env.SetInstanceData`. https://github.com/nodejs/node/blob/d615aeb7583b15bb5a8d1ec666ea29b8c7377455/src/js_native_api_v8.cc#L3201-L3205 does this delete the data?
- The final data would be `delete`d on environment cleanup.

Is my understanding correct?

Reply:
Eh, IIUC, it's node-addon-api's default finalizer deleted the data pointer: https://github.com/nodejs/node-addon-api/blob/main/napi.h#L208. Not quite intuitive tho... So no finalizer is required here.

Reply:
The `Delete` you point out only deleting the `RefBase` structure allocated in Node.js core, which would invoke the finalizer provided by node-addon-api.